core/auth/apis.go

```diff
@@ -659,6 +659,97 @@ func (a *Auth) CreateAdminAccount(authenticationType string, appID string, orgID
 	return newAccount, params, nil
 }
 
+// CreateAdminAccounts creates accounts for a new admin user
+func (a *Auth) CreateAdminAccounts(partialAccount []model.PartialAccount, creatorPermissions []string, clientVersion *string, l *logs.Log) ([]model.Account, map[string]interface{}, error) {
+	for _, p := range partialAccount {
+		if p.AuthType != AuthTypeOidc && p.AuthType != AuthTypeEmail && !strings.HasSuffix(p.AuthType, "_oidc") {
+			return nil, nil, errors.ErrorData(logutils.StatusInvalid, "auth type", nil)
+		}
+
+		// check if the provided auth type is supported by the provided application and organization
+		authType, appOrg, err := a.validateAuthTypeForAppOrg(p.AuthType, p.AppID, p.OrgID)
+		if err != nil {
+			errors.WrapErrorAction(logutils.ActionValidate, model.TypeAuthType, nil, err)
+		}
+
+		// create account
+		var newAccounts []model.Account
+		var params map[string]interface{}
+
+		transaction := func(context storage.TransactionContext) error {
```

Comment:
You should have one transaction for all operations but not every loop to have its own transaction.